lib/src/view/tournament/tournament_screen.dart

```diff
@@ -0,0 +1,489 @@
+            return BoardThumbnail(
+              size: boardSize,
+              fen: featuredGame.fen,
+              orientation: featuredGame.orientation,
+              header: const Shimmer(child: LoadingPlayerWidget()),
+              footer: const Shimmer(child: LoadingPlayerWidget()),
+            );
+        }
+      },
+    );
+  }
+}
+
+class _FeaturedGamePlayer extends StatelessWidget {
+  const _FeaturedGamePlayer({required this.game, required this.player, required this.side});
+
+  final PlayableGame game;
+  final FeaturedPlayer player;
+  final Side side;
+
+  @override
+  Widget build(BuildContext context) {
+    final activeClockSide = game.lastPosition.fullmoves > 1 ? game.lastPosition.turn : null;
+    return GamePlayer(
+      game: game,
+      side: side,
+      clock:
+          game.clock != null
+              ? CountdownClockBuilder(
+                key: key,
+                timeLeft: game.clockOf(side)!,
+                delay: game.clock!.lag ?? const Duration(milliseconds: 10),
+                clockUpdatedAt: game.clock!.at,
+                active: activeClockSide == side,
+                builder: (context, timeLeft) {
+                  return Clock(timeLeft: timeLeft, active: activeClockSide == side);
+                },
+              )
+              : null,
+    );
+  }
+}
+
+class _BottomBar extends ConsumerStatefulWidget {
+  const _BottomBar(this.state);
+
+  final TournamentState state;
+
+  @override
+  ConsumerState<_BottomBar> createState() => _BottomBarState();
+}
+
+class _BottomBarState extends ConsumerState<_BottomBar> {
+  bool joinOrLeaveInProgress = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final isLoggedIn = ref.watch(authSessionProvider)?.user.id != null;
+
+    ref.listen(
+      tournamentControllerProvider(widget.state.id).select((value) => value.valueOrNull?.joined),
+      (prevJoined, joined) {
+        if (prevJoined != joined) {
+          setState(() {
+            joinOrLeaveInProgress = false;
+          });
+        }
+      },
+    );
+
+    return PlatformBottomBar(
+      children: [
+        if (isLoggedIn)
+          joinOrLeaveInProgress
+              ? const Center(child: CircularProgressIndicator())
+              : BottomBarButton(
+                label: widget.state.joined ? context.l10n.pause : context.l10n.join,
+                icon: widget.state.joined ? Icons.pause : Icons.play_arrow,
+                showLabel: true,
+                onTap:
+                    widget.state.canJoin
+                        ? () {
+                          ref
+                              .read(tournamentControllerProvider(widget.state.id).notifier)
+                              .joinOrPause();
+                          setState(() {
+                            joinOrLeaveInProgress = true;
+                          });
+                        }
+                        : null,
+              )
+        else
+          BottomBarButton(
```

Comment:
Do we want to be able to sign in directly from here? I'm not sure, but why not.

Reply:
Yeah, the website and old app do it, and also right now the bottom bar would be empty otherwise :D